src/Magento/FunctionalTestingFramework/Util/TestGenerator.php

```diff
@@ -1267,7 +1267,16 @@ public function generateStepsPhp($actionObjects, $generationScope = TestGenerato
                         $actionObject->getActionOrigin()
                     )[0];
                     $argRef = "\t\t\$";
-                    $argRef .= str_replace(ucfirst($fieldKey), "", $stepKey) . "Fields['{$fieldKey}'] = ${input};\n";
+
+                    preg_match_all("/{{_CREDS\.([\w]+)}}/", "${input}", $matches);
```

Comment:
`preg` + conditional seems unecessary, see `resolveAllRuntimeReferences()` function on line 1877 and use it like:
`$input = $this->resolveAllRuntimeReferences([$input])[0]`

Reply:
Brilliant! Much cleaner!